File: shared/src/main/java/io/github/zwieback/familyfinance/util/StringUtils.java
package io.github.zwieback.familyfinance.util;

import android.support.annotation.Nullable;
import android.text.TextUtils;

public final class StringUtils {

    public static final String EMPTY = "";
    public static final String UNDEFINED = "undefined";

    public static boolean isTextEmpty(@Nullable CharSequence text) {
        return TextUtils.isEmpty(text);
    }

    public static boolean isTextNotEmpty(@Nullable CharSequence text) {
        return !isTextEmpty(text);
    }

    private StringUtils() {
    }
}

File: shared/src/main/java/io/github/zwieback/familyfinance/util/StringUtils.kt
package io.github.zwieback.familyfinance.util

object StringUtils {

    const val EMPTY = ""
    const val UNDEFINED = "undefined"
    const val QUESTION = "?"

    fun deleteLastChar(text: String?): String {
        return if (text?.isNotEmpty() == true) {
            text.substring(0, text.length - 1)
        } else {
            EMPTY
        }
    }

    fun addChar(text: String?, character: String): String {
        return if (text?.isNotEmpty() == true) {
            text + character
        } else {
            character
        }
    }

    fun addUniqueChar(text: String?, character: String): String {
        return if (text?.isNotEmpty() == true) {
            if (text.contains(character)) {
                text
            } else {
                text + character
            }
        } else {
            character
        }
    }
}
